### DISCUSSION TITLE
plot dengan agregasi lebih dari satu kolom target

### DISCUSSION THREAD

**Discussion by manuelatito** [teamalgoritma/community#894] [Q&A] [answered]
Dear Team Algoritma,

Melanjutkan poin pertanyaan saya untuk perbaikan plot pada LBB DV saya yang terdapat pada link berikut ini: https://rpubs.com/manuelatito/lbb_dv_tito
Terkait kondisi tersebut, ada pertanyaan terkait plot pada LBB DV yang ingin saya buat, namun saya sedikit kesusahan dalam membuatnya karena kondisi data saya. Untuk itu mohon bantuannya terkait hal ini.
Sebelumnya, saya ingin sedikit menggambarkan kondisi dari dataset saya terkait data polusi udara yang saya dapatkan dengan summary data sbb:

![image](https://github.com/user-attachments/assets/48ad3217-761d-46cc-88e1-3b38ee0a77a2)

Plot akhir yang saya buat adalah sebuah line plot untuk jenis polutan O3 dalam timeframe jam dalam 1 hari (pukul 00-23), dengan legend tahun, yaitu tahun 2017, 2018 dan 2019. Sehingga menghasilkan 3 line plot pada 1 layer seperti gambar berikut:

![image](https://github.com/user-attachments/assets/4b862ba7-5854-4792-8cb8-8a3dac9b3b7a)

Namun sebetulnya, bukan plot seperti itu yang saya inginkan.

Saya ingin membandingkan lebih dari 1 jenis polutan dalam plot saya (misal O3, SO2 dan NO2), sehingga 3 line plot yang dibuat adalah berdasarkan nilai dari 3 jenis polutan tersebut. 
Namun problemnya, 3 data polutan tersebut berada pada kolom yang berbeda yang seharusnya dilakukan aggregasi (dalam hal ini saya menggunakan mean (rata-rata)), sedangkan saya hanya tahu caranya melakukan agregasi dengan target 1 kolom saja. Berikut aggregasi yang saya buat untuk plot di atas.

![image](https://github.com/user-attachments/assets/f68b4269-e9b2-4bbd-8369-bc26df64c6db)

Pertanyaan saya, apakah memungkinkan jika saya ingin melakukan agregasi untuk menghitung 3 kolom (O3 , SO2 dan NO2) sehingga nantinya bisa saya tampilkan bersamaan dalam 1 canvas? Jika memungkinkan, mohon bantuan caranya bagaimana?
Berikut gambaran ekspektasi plot yang ingin saya buat:

![image](https://github.com/user-attachments/assets/22c1cd77-4ebe-4c96-97a2-fe969e833f00)

Selain itu, ada satu pertanyaan kecil yang juga mengganjal di pikiran saya, (mungkin tidak terlalu penting, namun saya hanya penasaran ingin tahu saja), yaitu terkait options scipen. Saya hanya ingin tahu saja, dalam beberapa materi ada yang menggunakan options(scipen = 999) ada juga yang menggunakan options(scipen = 9999). Sebenarnya apa yang membedakan kedua hal tersebut? Atau sebetulnya angkanya berapapun sama saja? Saya sudah mencoba googling terkait hal tersebut, namun belum mendapatkan jawaban yang jelas menurut pendapat saya. Mohon bantuannya dijelaskan.

Terima kasih atas bantuan dari team Algoritma untuk berkenan menjawab pertanyaan saya.

Best regards,
Tito

**Comment by Litaa** [comment, marked as answer]
Dear Ibu Tito, 

Terima kasih sudah mengajukan pertanyaan kepada kami.
Menanggapi pertanyaan Ibu tersebut di atas, apabila Ibu ingin membuat suatu visualisasi dengan informasi masing-masing line nya adalah jenis polutan, maka ibu harus melakukan reshaping terlebih dahulu pada data sebelum dilakukan proses aggregasi.
Proses reshaping ini dapat dengan menggunakan fungsi `pivot_longer`.
Apabila saat ini masing-masing polutan ditampung dalam kolom yang berbeda-beda, dengan menggunakan `pivot_longer()` data polutan akan dikelompokkan menjadi 2 kolom yang masing-masing berisi nama polutan (O2, NO2 & SO2) dan value nilai masing-masing polutan.
Berikut adalah referensi untuk pemanfaatan `pivot_longer` : https://tidyr.tidyverse.org/reference/pivot_longer.html

Setelah melakukan tahapan preparation data dengan `pivot_longer`, Ibu dapat melakukan aggregasi untuk setiap informasi tahunnya. Namun dapat diperhatikan kembali, apabila data hanya memuat sedikit informasi tahun (eg : Hanya tahun 2017, 2018 dan 2019), hasil line plot tidak akan terlalu berliku seperti yang terlihat pada gambar yang Ibu lampirkan. 

Mengenai option scipen, semakin besar nilai yang digunakan pada parameter, maka R akan semakin memaksa nilai yang dihasilkan tidak ditampilkan dalam bentuk notasi scientific. Sedangkan apabila parameter sciper bernilai negatif (eg: option(scipen = -5)), maka R akan semakin mudah untuk menampilkan angka dalam notasi scientific. 

Demikian informasi yang dapat kami sampaikan. Semoga membantu :)
Apabila ada pertanyaan lebih lanjut silahkan menghubungi kami kembali 😊🙏

**Reply by manuelatito** [reply]
Dear Kak Lita,

Terima kasih banyak atas jawaban dan ilmu barunya buat saya kak.
Wah saya sepertinya harus mempelajari dahulu fungsi `pivot_longer` ini yah kak. Terima kasih atas arahan dan referensi terkait `pivot_longer` tersebut. Namun saya butuh waktu untuk mempelajari dan mencobanya pada case lbb_dv saya ini, sehingga jika memang ada kesulitan kembali dalam mengimplementasikannya pada case saya, saya izin untuk menanyakan kembali di github di kemudian hari.

Oh itu maksudnya nilai parameter di option scipen. Baik, terima kasih banyak kak atas penjelasannya. Saya jadi semakin mengerti dan tidak penasaran lagi :)

Terima kasih banyak kak Lita atas jawabannya 😊🙏
